### PR TITLE
[DH-921] Add dit_adviser Interaction filter & default sort order by date desc.

### PR DESCRIPTION
**Pull request by elcct**
This adds missing `dit_adviser` Interaction filter and sets default sorting to `date:desc`.

### CODE REVIEW

File: datahub/search/interaction/tests/test_views.py [hunk 1]
```diff
@@ -184,15 +213,12 @@ def test_search_original_query(self, interactions, term):
...
-
-        for _ in range(10):
-            InteractionFactory(
-                kind=Interaction.KINDS.interaction,
-            )
-            service_deliveries.append(InteractionFactory(
-                kind=Interaction.KINDS.service_delivery,
-            ))
+        InteractionFactory.create_batch(10, kind=Interaction.KINDS.interaction),
+        service_deliveries = InteractionFactory.create_batch(
```
Comment (reupen) [inline datahub/search/interaction/tests/test_views.py]: Use ServiceDeliveryFactory?

File: datahub/search/interaction/tests/test_views.py [hunk 2]
```diff
@@ -77,6 +79,33 @@ def test_offset(self):
...
+            datetime(2017, 2, 4, 13, 15, 0),
+            datetime(2017, 1, 4, 11, 23, 10),
+            datetime(2017, 9, 29, 3, 25, 15),
+            datetime(2017, 7, 5, 11, 44, 33),
+            datetime(2017, 2, 1, 18, 15, 1),
+        )
+        date_iter = iter(dates)
+        InteractionFactory.create_batch(
+            len(dates),
+            date=factory.LazyFunction(lambda: next(date_iter))
```
Comment (reupen) [inline datahub/search/interaction/tests/test_views.py]: What about factory.Iterator?

File: datahub/search/interaction/tests/test_views.py (hunk 1 above)
Comment (elcct) [reply datahub/search/interaction/tests/test_views.py]: I should have looked for its existence. Thanks :)

File: datahub/search/interaction/tests/test_views.py (hunk 2 above)
Comment (elcct) [reply datahub/search/interaction/tests/test_views.py]: That's exactly what I was looking for :) Thanks! Fixed.

File: datahub/search/interaction/tests/test_views.py [hunk 3]
```diff
@@ -287,15 +315,17 @@ def test_filter_by_contact_id(self, setup_es):
...
-            ))
-            InteractionFactory(
-                contact=contacts[i]
-            )
+        name_str = partial(get_random_string, 16)
+
+        contacts = ContactFactory.create_batch(
+            10,
+            first_name=factory.LazyFunction(name_str),
+            last_name=factory.LazyFunction(name_str)
```
Comment (reupen) [inline datahub/search/interaction/tests/test_views.py]: I know it was like this before, but looks like we could use factory.Faker() here (with 'first_name'/'last_name') (in fact ContactFactory should be changed).

File: datahub/search/interaction/tests/test_views.py [hunk 4]
```diff
@@ -237,10 +260,14 @@ def test_filter_by_company_id(self, setup_es):
 
     def test_filter_by_company_name(self, setup_es):
         """Tests filtering interaction by company name."""
-        companies = []
-        for i in range(10):
-            companies.append(CompanyFactory(name=get_random_string(16)))
-            InteractionFactory(company=companies[i])
+        companies = CompanyFactory.create_batch(
+            10,
+            name=factory.LazyFunction(partial(get_random_string, 16))
```
Comment (reupen) [inline datahub/search/interaction/tests/test_views.py]: Similar to below (looks like CompanyFactory could use `factory.Faker('company')`, but could be fixed in a separate PR).

File: datahub/search/interaction/tests/test_views.py (hunk 3 above)
Comment (elcct) [reply datahub/search/interaction/tests/test_views.py]: Great idea. I changed ContactFactory.

File: datahub/search/interaction/tests/test_views.py (hunk 4 above)
Comment (elcct) [reply datahub/search/interaction/tests/test_views.py]: Also changed CompanyFactory, thanks :)

**Pull request merged**